src/compiletest/runtest.rs

```diff
@@ -1013,8 +1013,8 @@ fn check_expected_errors(revision: Option<&str>,
         expected_errors.iter()
                         .fold((false, false),
```

Comment:
Do it in two passes?


On Wed, Mar 16, 2016 at 12:25:23PM -0700, Corey Farwell wrote:
> > @@ -1013,8 +1013,8 @@ fn check_expected_errors(revision: Option<&str>,
> >          expected_errors.iter()
> >                          .fold((false, false),
> 
> Given this code block:
> 
> ```rust
> fn main() {
>   let i = 1; //~ NOTE: some note
>   let j = 2; //~ HELP: some help 
> }
> ```
> 
> The current implementation checks for `note` and `help` in one pass through. I'm not sure how it'd be possible using `any` unless we were expecting **both** `help` and `note` on the same line. I agree it's not clean, and I'm willing to clean it up, but would rather do that in a different PR.
> 
> ---
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub:
> https://github.com/rust-lang/rust/pull/32263/files/fb96cc8b5d83da8c32cdaf51a5f53edb74152024#r56398787